lang/python/python-package.mk

```diff
@@ -94,6 +94,9 @@ define Build/Compile/HostPyRunTarget
 		CXX="$(TARGET_CXX)" \
 		LD="$(TARGET_CC)" \
 		LDSHARED="$(TARGET_CC) -shared" \
+		F90="$(TARGET_CROSS)gfortran" \
+		AR="$(TARGET_AR)" \
+		RANLIB="$(TARGET_RANLIB)" \
```

Comment:
Would prefer to set these using `PYTHON[3]_PKG_SETUP_VARS`, unless there are other packages that also need this.

Reply:
Are you sure? Fortran seems pretty standard to me and is probably used in more scientific libraries. It should not break anything in cases where the Fortran compiler is not used.

As far as ar and ranlib are concerned it may be unnecessary on Linux (I don't think ARCH makes a difference for these binutils tools). On macOS ar is platform agnostic but ranlib is not (and the error one gets is not very enlightening). In any case it is safer to use proper cross-tools and adding it for all packages may save someone some headaches in the future.

Reply:
I've grepped packages in the base and packages (this one) feeds, not a single package uses Fortran.

I know you were developing with 18.06 which doesn't have [PYTHON[3]_PKG_SETUP_VARS](https://github.com/openwrt/packages/tree/master/lang/python#building), but if you were to submit a PR for master, I would suggest you use it.

Reply:
I think `AR` & `RANLIB` would be fine to be set here.
They look like they could be beneficial for other packages with macOS; they could be made part of the "python cross-build infrastructure".
Since I don't have a Mac machine, cross-compiling on a Mac for me is a black-box.

Whereas fortran, I personally don't mind; it looks more like a relic [I agree].
No strong opinion on it.

In general, I am fine with taking some risks every now-n-then.
This looks like one of those moments where things look weird enough that the first instinct is to resist them.

Well, this is my vote about this :)

Reply:
Yeah, I forgot that Fortran is a numpy-specific addition to distutils so it totally makes sense to move it to the numpy package using PYTHON_PKG_SETUP_VARS.
I think `AR` and `RANLIB` make more sense for packages other than numpy so I think it is a good idea to keep it.

Reply:
I'll defer to @commodo on adding `AR` and `RANLIB`.

But to beat a dead horse one last time: I believe in the [rule of three](https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)). If there are other existing packages need similar workarounds for `AR` and/or `RANLIB`, then I'd be happy to refactor this code into python-package.mk. But if there isn't, then we are adding code that has no immediate benefit but increases the potential for error (across all Python packages).

Reply:
i'm more inclined to stick with AR & RANLIB as proposed by @jpc

to some degree, I feel that some of these `TARGET_` options should be similar or identical to `TARGET_CONFIGURE_OPTS` as here https://github.com/openwrt/openwrt/blob/master/rules.mk#L302

macOS is a bit special sometimes during cross-building (especially when building OpenWrt) in the fact that it reveals weird stuff; and not many people try to build OpenWrt on macOS;
and i've also wasted time to fix mac-build stuff (when I had one around at work)

sometimes, I look at buildroot for inspiration (because OpenWrt is an older fork of buildroot), but that still doesn't offer any help in this AR + RANLIB discussion;

personally, i am willing to take a chance with this change, so if there's any breakage with it, we can probably revert;

one thing does come to mind here though: maybe i should buy a Mac as personal computer  ¯\_(ツ)_/¯